src/Common/src/TypeSystem/CodeGen/TargetDetails.CodeGen.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
+{
+    // Extension to TargetDetails related to code generation
+    partial class TargetDetails
+    {
+        public TargetDetails(TargetArchitecture architecture, TargetOS targetOS, TargetAbi abi, MaximumSimdVectorLength simdVectorLength)
+            : this(architecture, targetOS, abi)
+        {
+            MaximumSimdVectorLength = simdVectorLength;
+        }
+
+        public MaximumSimdVectorLength MaximumSimdVectorLength
+        {
+            get;
+        }
+    }
+
+    public enum MaximumSimdVectorLength
+    {
+        None,
+        VectorLength16,
+        VectorLength32,
+    }
```

Comment:
I don't like these enum names. What does the 16 or 32 refer to? Bytes? Bits? Elements? In general, I've seen vectors named based on bit length more than on byte length. (For instance, AVX512) I'd prefer something like Vector64Bit